src/libsync/propagatedownloadzsync.cpp

```diff
@@ -187,6 +188,11 @@ void GETFileZsyncJob::seedFailed(QString errorString)
 
     qCCritical(lcZsyncGet) << _errorString;
 
+    /* delete remote zsync file */
+    QUrl zsyncUrl = Utility::concatUrlPath(_propagator->account()->zsyncUrl(),
+                                           _propagator->_remoteFolder + _item->_file);
+    (new DeleteJob(_propagator->account(), zsyncUrl, this))->start();
```

Comment:
Hm, does the parent of this delete job live long enough? If this is destroyed quickly after this function finishes the DeleteJob would never run.

Reply:
When I tested it, it did live long enough ...

Reply:
@guruz I expect this would be safer if the DeleteJob didn't have a parent at all. Do you agree? (also for the other DeleteJob below) The DeleteJob will autodelete itself when it's done anyway.